src/main/scala/firrtl/options/ExecutionOptionsManager.scala

```diff
@@ -0,0 +1,63 @@
+// See LICENSE for license details.
+
+package firrtl.options
+
+import firrtl.AnnotationSeq
+
+import scopt.OptionParser
+
+import java.util.ServiceLoader
+
+/** Indicate an error in [[firrtl.options]]
+  * @param msg a message to print
+  */
+case class OptionsException(msg: String, cause: Throwable = null) extends Exception(msg, cause)
+
+/** A utility for working with command line options
+  * @param applicationName the application associated with these command line options
+  */
+class ExecutionOptionsManager(val applicationName: String) {
```

Comment:
@seldridge You suggested we might change this name and the more I think about it the more I agree. I think it will be confusing to have two of them, the name has always been a mouthful and there's some question about what execution means to me too. Do you have any good names up your sleeve for this (after your triumphant selection of the word Stage). It's not a deal breaker if you'd like to keep it in order to get this stuff over, but I vote for changing it now. Let me know.

Reply:
Yeah, this name is clearly broken (and should probably be fixed in [F764.5](https://github.com/freechipsproject/firrtl/pull/879)). In the follow-on commit, [F764.7](https://github.com/freechipsproject/firrtl/pull/920) I define `FirrtlCli` and `LoggerCli` which are self-typed as `ExecutionOptionsManager`. It made sense back when this was a store of execution options, but this patch series moves the options into the annotatinos.

So, what is this doing now?
1. Defines the name of the program presented to the user via a CLI
2. Defines a parser
3. Registers options
4. Defines a `parse` method to convert `Array[String] -> AnnotationSeq` using the parser

With that in mind, how about:
- `CommandLineInterface`
- `CLI`
- `Console`
- `Shell`

`CommandLineInterface` is clean as it describes what's going on, but I kind of like the shorter `Shell`. Maybe open it up for some quality developer bikeshedding.

Reply:
Settled on `Shell` and Updated F764.5 to use that.